lib/osf-components/addon/components/move-file-modal/template.hbs

```diff
@@ -0,0 +1,155 @@
+<OsfDialog
+    @onOpen={{this.onOpen}}
+    @onClose={{this.onClose}}
+    @fixedWidth={{true}}
+    as |dialog|
+>
+    <dialog.trigger>
+        <Button
+            @layout='fake-link'
+            {{on 'click' dialog.open}}
+        >
+            <FaIcon
+                @icon='arrow-right'
+            />
+            {{t 'general.move'}}
+        </Button>
+    </dialog.trigger>
+    <dialog.heading>
+        {{t 'osf-components.move_file_modal.header'}}
+    </dialog.heading>
+    <dialog.main>
+        {{#if this.isMoving}}
+            <h3>{{t 'osf-components.move_file_modal.moving_files'}}</h3>
+            {{#each this.fileMoveTasks as |task| }}
+                {{#let (get task.args 0) as |file| }}
+                    {{#if task.isRunning}}
+                        <p>
+                            {{t 'osf-components.move_file_modal.moving_file' fileName=file.name}}
+                            <FaIcon @icon='spinner' @pulse={{true}} />
+                        </p>
+                    {{else if task.isSuccessful}}
+                        {{t 'osf-components.move_file_modal.success' fileName=file.name}}
+                        <FaIcon @icon='check' />
+                    {{else if task.isError}}
+                        {{!-- TODO: add retry logic --}}
+                        {{t 'osf-components.move_file_modal.error' fileName=file.name}}
+                        {{task.error.responseText}}
+                        <Button>
+                            {{t 'osf-components.move_file_modal.retry'}}
+                        </Button>
+                        <Button>
+                            {{t 'osf-components.move_file_modal.replace'}}
+                        </Button>
+                    {{/if}}
+                {{/let}}
+            {{/each}}
+        {{!-- {{else if this.moveDone}}
+            need something if it works and something if it doesn't
+            <h3>{{t 'osf-components.move_file_modal.done'}}</h3> --}}
+        {{else}}
+            <h3>{{t 'osf-components.move_file_modal.select_destination'}}</h3>
+            <div>
+                Files to move:
+                {{#each @filesToMove as |file|}}
+                    {{file.name}}
+                {{/each}}
```

Comment:
So if someone is moving 1000 files, there'll be a list of 1000 filenames here?

Reply:
Oh, this is for my goldfish brain to remember which files I'm moving while I test it out locally 🐟

Reply:
Although, it may be nice to see which files are being moved. Maybe I could add a little bit of logic here so it says something like: "Moving abc, def, and {number} other files"